### PR TITLE
feat: WT-2055 need more eth drawer

### PR DESCRIPTION
**Pull request by ZacharyCouchman**
[WT-2055](https://immutable.atlassian.net/browse/WT-2055)

Adding Need More Eth drawer for claim withdrawal.

image

[WT-2055]: https://immutable.atlassian.net/browse/WT-2055?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/checkout/widgets-lib/src/components/Transactions/NotEnoughEthToWithdraw.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            size="small"
+            sx={contentTextStyles}
+            testId="not-enough-gas-heading"
+          >
+            {/* eslint-disable-next-line max-len */}
+            {`${t('drawers.notEnoughEthWithdrawal.content.heading1')} ETH ${t('drawers.notEnoughEthWithdrawal.content.heading2')}`}
+          </Heading>
+          <Body sx={contentTextStyles}>
+            {/* eslint-disable-next-line max-len */}
+            {`${t('drawers.notEnoughEthWithdrawal.content.body1')} ETH ${t('drawers.notEnoughEthWithdrawal.content.body2')}`}
```
Comment (imx-mikhala) [inline packages/checkout/widgets-lib/src/components/Transactions/NotEnoughEthToWithdraw.tsx:48]: should wew use the constant `${ETH_TOKEN_SYMBOL}` for this?

File: packages/checkout/widgets-lib/src/widgets/bridge/views/ClaimWithdrawal.tsx [hunk 2]
```diff
@@ -169,13 +190,12 @@ export function ClaimWithdrawal({ transaction }: ClaimWithdrawalProps) {
...
-      //   provider: providerToUse,
-      //   transaction: withdrawalResponse.unsignedTx,
-      // });
-
+      const response = await checkout.sendTransaction({
+        provider: providerToUse,
+        transaction: withdrawalResponse.unsignedTx,
+      });
+      // eslint-disable-next-line no-console
+      console.log(response);
```
Comment (imx-mikhala) [inline packages/checkout/widgets-lib/src/widgets/bridge/views/ClaimWithdrawal.tsx:198]: is this console.log intentionally committed?

File: packages/checkout/widgets-lib/src/components/Transactions/NotEnoughEthToWithdraw.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  onChangeAccount: () => void,
+}
+
+export function NotEnoughEthToWithdraw({
+  visible,
+  onClose,
+  onChangeAccount,
+}: NotEnoughEthToWithdrawProps) {
+  const { t } = useTranslation();
+  const ethLogo = `${CHECKOUT_CDN_BASE_URL[Environment.PRODUCTION]}/v1/blob/img/tokens/eth.svg`;
```
Comment (imx-mikhala) [inline packages/checkout/widgets-lib/src/components/Transactions/NotEnoughEthToWithdraw.tsx:24]: using `PRODUCTION` env here - can we get the env instead from the checkout object on context

**Review by dreamoftrees — DISMISSED** [review]
Nice work @ZacharyCouchman 👍

File: packages/checkout/widgets-lib/src/components/Transactions/NotEnoughEthToWithdraw.tsx (hunk 3 above)
Comment (ZacharyCouchman) [reply packages/checkout/widgets-lib/src/components/Transactions/NotEnoughEthToWithdraw.tsx:24]: Yeah to be honest I thought this was more work than what it was worth. I doubt that we are going to change image here based on environment, that's why I left it out.

Comment (ZacharyCouchman) [reply packages/checkout/widgets-lib/src/components/Transactions/NotEnoughEthToWithdraw.tsx:24]: But then again, we may decide to update images in the future and we may want to test them in sandbox first? what do you think?

File: packages/checkout/widgets-lib/src/widgets/bridge/views/ClaimWithdrawal.tsx (hunk 2 above)
Comment (ZacharyCouchman) [reply packages/checkout/widgets-lib/src/widgets/bridge/views/ClaimWithdrawal.tsx]: I think this is outdated now. Should be removed

File: packages/checkout/widgets-lib/src/components/Transactions/NotEnoughEthToWithdraw.tsx (hunk 3 above)
Comment (imx-mikhala) [reply packages/checkout/widgets-lib/src/components/Transactions/NotEnoughEthToWithdraw.tsx:24]: i think testing symbols in sandbox is a good reason to use env from checkout obj

File: packages/checkout/widgets-lib/src/components/NotEnoughGas/NotEnoughGas.tsx [hunk 4]
```diff
@@ -65,11 +64,15 @@ NotEnoughGasProps) {
     >
       <Drawer.Content>
         <Box testId="not-enough-gas-bottom-sheet" sx={containerStyles}>
-          <FramedImage
-            imageUrl={tokenSymbol === 'ETH' ? ethLogo : imxLogo}
-            circularFrame
-            sx={tokenSymbol === ETH_TOKEN_SYMBOL ? ethLogoStyles : imxLogoStyles}
+          <CloudImage
+            imageUrl={
+              tokenSymbol === 'ETH'
```
Comment (imx-mikhala) [inline packages/checkout/widgets-lib/src/components/NotEnoughGas/NotEnoughGas.tsx:69]: could use the ETH constant here too

**Review by dreamoftrees — DISMISSED** [review]
👍

**Review by imx-mikhala — DISMISSED** [review]

**Review by imx-mikhala — APPROVED** [review]

**Review by dreamoftrees — DISMISSED** [review]
👍

**Review by dreamoftrees — APPROVED** [review]
👍

File: packages/checkout/widgets-lib/src/components/NotEnoughGas/NotEnoughGas.tsx [hunk 5]
```diff
@@ -26,21 +26,20 @@ type NotEnoughGasProps = {
...
-NotEnoughGasProps) {
+}: NotEnoughGasProps) {
   const { content, buttons } = text.drawers.notEnoughGas;
 
   const [isCopied, setIsCopied] = useState(false);
-
-  const ethLogo = 'https://design-system.immutable.com/hosted-for-ds/currency-icons/currency--eth.svg';
-  const imxLogo = 'https://design-system.immutable.com/hosted-for-ds/currency-icons/currency--imx.svg';
+  const ethLogo = `${CHECKOUT_CDN_BASE_URL[environment]}/v1/blob/img/tokens/eth.svg`;
+  const imxLogo = `${CHECKOUT_CDN_BASE_URL[environment]}/v1/blob/img/tokens/imx.svg`;
```
Comment (andrearampin) [inline packages/checkout/widgets-lib/src/components/NotEnoughGas/NotEnoughGas.tsx:42]: we should really try to move these to an utility function.

File: packages/checkout/widgets-lib/src/components/Transactions/NotEnoughEthToWithdraw.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            size="small"
+            sx={contentTextStyles}
+            testId="not-enough-gas-heading"
+          >
+            {/* eslint-disable-next-line max-len */}
+            {`${t('drawers.notEnoughEthWithdrawal.content.heading1')} ${ETH_TOKEN_SYMBOL} ${t('drawers.notEnoughEthWithdrawal.content.heading2')}`}
+          </Heading>
+          <Body sx={contentTextStyles}>
+            {/* eslint-disable-next-line max-len */}
+            {`${t('drawers.notEnoughEthWithdrawal.content.body1')} ${ETH_TOKEN_SYMBOL} ${t('drawers.notEnoughEthWithdrawal.content.body2')}`}
```
Comment (andrearampin) [inline packages/checkout/widgets-lib/src/components/Transactions/NotEnoughEthToWithdraw.tsx:50]: Why do we need this `eslint` rules? Cannot we simply break the line?

File: packages/checkout/widgets-lib/src/lib/constants.ts [hunk 7]
```diff
@@ -105,3 +105,5 @@ export const PASSPORT_URL = {
   [Environment.SANDBOX]: 'https://passport.sandbox.immutable.com/',
   [Environment.PRODUCTION]: 'https://passport.immutable.com/',
 };
+
+export const WITHDRAWAL_CLAIM_GAS_LIMIT = 91000;
```
Comment (andrearampin) [inline packages/checkout/widgets-lib/src/lib/constants.ts:109]: Why is this needed? Or, shouldn't this be part of the Bridge SDKs?

Comment (ZacharyCouchman) [reply packages/checkout/widgets-lib/src/lib/constants.ts:109]: I've added it in case our estmateGas call fails.

File: packages/checkout/widgets-lib/src/components/Transactions/NotEnoughEthToWithdraw.tsx (hunk 6 above)
Comment (ZacharyCouchman) [reply packages/checkout/widgets-lib/src/components/Transactions/NotEnoughEthToWithdraw.tsx:50]: It's easier to see that they should be formatted with a space between the items if they're all on one line

**Review by imx-mikhala — APPROVED** [review]

**Review by dreamoftrees — APPROVED** [review]
👍

File: packages/checkout/widgets-lib/src/lib/constants.ts (hunk 7 above)
Comment (andrearampin) [reply packages/checkout/widgets-lib/src/lib/constants.ts:109]: is that limit the same for Testnet and Mainnet?

File: packages/checkout/widgets-lib/src/widgets/bridge/views/ClaimWithdrawal.tsx [hunk 8]
```diff
@@ -227,9 +253,11 @@ export function ClaimWithdrawal({ transaction }: ClaimWithdrawalProps) {
               size="large"
               variant="primary"
               disabled={loading}
-              onClick={(txProcessing || loading) ? () => { } : handleWithdrawalClaimClick}
+              onClick={() => ((txProcessing || loading)
+                ? undefined
+                : handleWithdrawalClaimClick({ forceChangeAccount: true }))}
             >
-              {loading ? (
+              {loading || txProcessing ? (
```
Comment (andrearampin) [inline packages/checkout/widgets-lib/src/widgets/bridge/views/ClaimWithdrawal.tsx:260]:
```suggestion
              {(loading || txProcessing) ? (
```
shouldn't this be?
